include/zephyr/sys/device_mmio.h

```diff
@@ -63,13 +63,13 @@ struct z_device_mmio_rom {
 
 #define Z_DEVICE_MMIO_ROM_INITIALIZER(node_id) \
 	{ \
-		.phys_addr = DT_REG_ADDR(node_id), \
+		.phys_addr = UINT32_C(DT_REG_ADDR(node_id)), \
```

Comment:
tempting to have this as part of `DT_REG_ADDR` (and potentially `DT_REG_SIZE` too) macro itself but that would be a substantial change, definitely worth its own PR, I propose to patch this one up for now. Somewhat surprised CI does not hit more cases of this warning though.

Reply:
it's a quick fix, that's pretty much what I meant in the comment above

Reply:
Please create an issue for this then, if its meant to be revisited

Reply:
There already exist an issue:
https://github.com/zephyrproject-rtos/zephyr/issues/53505
I gave it a shot, but gave up:
https://github.com/zephyrproject-rtos/zephyr/pull/70761

Reply:
@jeppenodgaard what was wrong with that? CI errors? I was considering giving it a go as well

Reply:
@fabiobaltieri see this closing comment:
https://github.com/zephyrproject-rtos/zephyr/pull/70761#issuecomment-2295972704